### PR TITLE
Super naive biased sampler version

### PR DESCRIPTION
**Pull request by bischtob**
This PR
- adds a biased EM sampler
- adds a test that ensures when the bias function returns zero or the bias is set to nothing that we recover our original EM sampler. I kept our original EM sampler
- adds an event probability plot to utils analysis
- moves the read from hdf5 and drop to hdf5 functions to utils_etl 
- renames store_load_samples to generate_samples. This generates biased samples or not based on kwargs and the Experiment toml [relying on the fact that when k_bias = 0 or bias = nothing, the `ld_sampler` is the same as the original EM sampler].
- Modifies the correlated OU exp toml to have bias terms 
- Adds an analysis script for biased samples. this is sufficiently different from our quick analysis script (to test that the generative model works in the first place) that I felt it was OK to have a duplicate. we could try and make it slick, and reduce some code duplication, but for now it seemed fine to keep them separate.

Note that in the future it would be nice to have `generate_samples` for all the examples, but right now this depends on example specific details. so, TBD on how to do that. 

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/sampling.jl [hunk 1]
```diff
@@ -43,6 +43,57 @@ function Euler_Maruyama_sampler(model::CliMAgen.AbstractDiffusionModel,
     return x
 end
 
+"""
+    Euler_Maruyama_ld_sampler(model::CliMAgen.AbstractDiffusionModel,
```
Comment (kmdeck) [inline src/sampling.jl:47]: can update doc string to include description of bias argument

File: src/sampling.jl [hunk 2]
```diff
@@ -43,6 +43,57 @@ function Euler_Maruyama_sampler(model::CliMAgen.AbstractDiffusionModel,
...
+    @showprogress "Euler-Maruyama Sampling" for time_step in time_steps
+        batch_time_step = fill!(similar(init_x, size(init_x)[end]), 1) .* time_step
+        g = CliMAgen.diffusion(model, batch_time_step)
+        if forward
+            x = x .+ sqrt(Δt) .* CliMAgen.expand_dims(g, 3) .* randn!(similar(x))
+        else
+            if bias isa Nothing
+                score = CliMAgen.score(model, x, batch_time_step; c=c)
+            else
+                score = CliMAgen.score(model, x, batch_time_step; c=c) .+ bias
```
Comment (kmdeck) [inline src/sampling.jl:88]: double checking: bias should be k*grad_x A(x)

Comment (kmdeck) [reply src/sampling.jl:88]: In this case it is OK, but to be more general, we should pass in a bias that is a function which takes (x) and returns something of the same size. Then here we also evaluated bias(x)

File: examples/correlated_ou_1d/analysis_ld.jl [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    xtrain = xtrain[:, :, :, 1:nsamples]
+
+    if make_samples
+        # set up model
+        checkpoint_path = joinpath(savedir, "checkpoint.bson")
+        BSON.@load checkpoint_path model model_smooth opt opt_smooth
+        model = device(model)
+        
+        # set up bias for space-time mean
+        bias = zeros(FT, n_pixels, n_time, inchannels)
```
Comment (kmdeck) [inline examples/correlated_ou_1d/analysis_ld.jl:79]: looks good! d(mean(x))dx = 1.

The indexing of `bias` indicates that we want to bias the middle half of the pixels in time and space.

**Review by bischtob — COMMENTED** [review]
I checked the files! i think you need to review once more.

**Review by kmdeck — APPROVED** [review]
🚀

**Pull request merged**